### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by F2had**

### CODE REVIEW

File: lib/modules/settings/screens/current_location_picker.dart [hunk 1]
```diff
@@ -70,41 +94,36 @@ class _CurrentLocationPickerState extends State<CurrentLocationPicker> {
...
+      }
+
+      var response = await getCoordinatesFromAddress(address);
+      var location = json.decode(response.body);
+      _updateUserLocation(context, location['latitude'] as double,
+          location['longitude'] as double);
     }
   }
 
+  _geoCodingAddress(String address) async {
```
Comment (Amjad50) [inline lib/modules/settings/screens/current_location_picker.dart:116]: address is not being used

File: lib/modules/settings/screens/current_location_picker.dart [hunk 2]
```diff
@@ -70,41 +94,36 @@ class _CurrentLocationPickerState extends State<CurrentLocationPicker> {
...
+    //TODO: Add better error handling
+    Location locations = await locationFromAddress(
```
Comment (Amjad50) [inline lib/modules/settings/screens/current_location_picker.dart:118]: Small nitpick: location

File: lib/modules/settings/screens/current_location_picker.dart [hunk 3]
```diff
@@ -70,41 +94,36 @@ class _CurrentLocationPickerState extends State<CurrentLocationPicker> {
...
+        _updateUserLocation(context, location.latitude, location.longitude);
+        return;
+      } catch (e) {
+        debugPrint(e.toString());
+      }
+
+      var response = await getCoordinatesFromAddress(address);
+      var location = json.decode(response.body);
+      _updateUserLocation(context, location['latitude'] as double,
+          location['longitude'] as double);
```
Comment (Amjad50) [inline lib/modules/settings/screens/current_location_picker.dart:112]: In here, we take the values of "country", "state",.... Then get the lat-lon, then use those to retrieve the "country", "state", values from Placemark API (in _updateUserLocation).
I think this unnecessary.

Comment (F2had) [reply lib/modules/settings/screens/current_location_picker.dart:112]: To make use of [placemarkFromCoordinates](https://github.com/NedaaApp/nedaa/pull/15/files/bf9fab9f6a631e6b2de5994be08de34f4bc9258b#diff-8cf13d88a7f89dd55e33e3d3512e50b7092aaac40cf9a0f9050986aac9ecdf17) to translate the country, state and city based on the current locale

**Review by Amjad50 — APPROVED** [review]

**Pull request merged**